Cannot compile pc-nrfjprog-js on Arch Linux

When I try to install pc-nrfjprog-js as a dependency, the compilation fails:

<details>
  <summary>error log</summary>

    > pc-nrfjprog-js@1.0.0 install /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js
    > node build.js

    info DIST Downloading distribution files.
    http DIST       - http://nodejs.org/dist/v6.11.3/SHASUMS256.txt
    http DIST       - http://nodejs.org/dist/v6.11.3/node-v6.11.3-headers.tar.gz
    info CMD CLEAN
    info RUN cmake -E remove_directory "/home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/build"
    info CMD CONFIGURE
    WARN invalid config loglevel="notice"
    info RUN cmake "/home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js" --no-warn-unused-cli -DCMAKE_BUILD_TYPE="Release" -DCMAKE_LIBRARY_OUTPUT_DIRECTORY="/home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/build/Release" -DCMAKE_JS_INC="/home/m/.cmake-js/node-x64/v6.11.3/include/node;/home/m/git/cloud-iris/gateway/js/common/node_modules/nan" -DNODE_RUNTIME="node" -DNODE_RUNTIMEVERSION="6.11.3" -DNODE_ARCH="x64" -DCMAKE_C_COMPILER="clang" -DCMAKE_CXX_COMPILER="clang++" -DCMAKE_CXX_FLAGS="-std=c++11"
    Not searching for unused variables given on the command line.
    -- The C compiler identification is Clang 4.0.1
    -- The CXX compiler identification is Clang 4.0.1
    -- Check for working C compiler: /usr/bin/clang
    -- Check for working C compiler: /usr/bin/clang -- works
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Check for working CXX compiler: /usr/bin/clang++
    -- Check for working CXX compiler: /usr/bin/clang++ -- works
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- 
    -- Configuring done
    -- Generating done
    -- Build files have been written to: /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/build
    info CMD BUILD
    info RUN cmake --build "/home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/build" --config Release
    Scanning dependencies of target pc-nrfjprog-js
    [ 10%] Building CXX object CMakeFiles/pc-nrfjprog-js.dir/src/common.cpp.o
    warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
    1 warning generated.
    [ 20%] Building CXX object CMakeFiles/pc-nrfjprog-js.dir/src/nrfjprog_helpers.cpp.o
    warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
    1 warning generated.
    [ 30%] Building CXX object CMakeFiles/pc-nrfjprog-js.dir/src/nrfjprogjs.cpp.o
    warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
    In file included from /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.cpp:41:
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.h:49:14: error: no type named 'function' in namespace 'std'
    typedef std::function<Baton*(Nan::NAN_METHOD_ARGS_TYPE, int&)> parse_parameters_function_t;
            ~~~~~^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.h:49:22: error: expected unqualified-id
    typedef std::function<Baton*(Nan::NAN_METHOD_ARGS_TYPE, int&)> parse_parameters_function_t;
                        ^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.h:50:14: error: no type named 'function' in namespace 'std'
    typedef std::function<nrfjprogdll_err_t(Baton*, Probe_handle_t)> execute_function_t;
            ~~~~~^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.h:50:22: error: expected unqualified-id
    typedef std::function<nrfjprogdll_err_t(Baton*, Probe_handle_t)> execute_function_t;
                        ^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.h:51:14: error: no type named 'function' in namespace 'std'
    typedef std::function<returnType(Baton*)> return_function_t;
            ~~~~~^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.h:51:22: error: expected unqualified-id
    typedef std::function<returnType(Baton*)> return_function_t;
                        ^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.h:88:36: error: unknown type name 'parse_parameters_function_t'
                                const parse_parameters_function_t parse,
                                    ^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.h:89:36: error: unknown type name 'execute_function_t'
                                const execute_function_t execute,
                                    ^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.h:90:36: error: unknown type name 'return_function_t'
                                const return_function_t ret,
                                    ^
    In file included from /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.cpp:43:
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs_batons.h:84:5: error: unknown type name 'execute_function_t'
        execute_function_t executeFunction;
        ^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs_batons.h:85:5: error: unknown type name 'return_function_t'
        return_function_t returnFunction;
        ^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.cpp:110:61: error: unknown type name 'parse_parameters_function_t'
    void nRFjprog::CallFunction(Nan::NAN_METHOD_ARGS_TYPE info, parse_parameters_function_t parse, execute_function_t execute, return_function_t ret, const bool hasSerialNumber)
                                                                ^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.cpp:110:96: error: unknown type name 'execute_function_t'
    void nRFjprog::CallFunction(Nan::NAN_METHOD_ARGS_TYPE info, parse_parameters_function_t parse, execute_function_t execute, return_function_t ret, const bool hasSerialNumber)
                                                                                                ^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.cpp:110:124: error: unknown type name 'return_function_t'
    void nRFjprog::CallFunction(Nan::NAN_METHOD_ARGS_TYPE info, parse_parameters_function_t parse, execute_function_t execute, return_function_t ret, const bool hasSerialNumber)
                                                                                                                            ^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.cpp:414:5: error: unknown type name 'parse_parameters_function_t'
        parse_parameters_function_t p = [&] (Nan::NAN_METHOD_ARGS_TYPE parameters, int &argumentCount) -> Baton* {
        ^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.cpp:417:16: error: cannot initialize return object of type 'Baton *' with an lvalue of type 'GetDllVersionBaton *'
            return baton;
                ^~~~~
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.cpp:420:5: error: use of undeclared identifier 'execute_function_t'; did you mean 'ExecuteFunction'?
        execute_function_t e = [&] (Baton *b, Probe_handle_t probe) -> nrfjprogdll_err_t {
        ^~~~~~~~~~~~~~~~~~
        ExecuteFunction
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.cpp:207:16: note: 'ExecuteFunction' declared here
    void nRFjprog::ExecuteFunction(uv_work_t *req)
                ^
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.cpp:420:23: error: expected ';' after expression
        execute_function_t e = [&] (Baton *b, Probe_handle_t probe) -> nrfjprogdll_err_t {
                        ^
                        ;
    /home/m/git/cloud-iris/gateway/js/common/node_modules/pc-nrfjprog-js/src/nrfjprogjs.cpp:420:24: error: use of undeclared identifier 'e'
        execute_function_t e = [&] (Baton *b, Probe_handle_t probe) -> nrfjprogdll_err_t {
                        ^
    fatal error: too many errors emitted, stopping now [-ferror-limit=]
    1 warning and 20 errors generated.
    make[2]: *** [CMakeFiles/pc-nrfjprog-js.dir/build.make:111: CMakeFiles/pc-nrfjprog-js.dir/src/nrfjprogjs.cpp.o] Error 1
    make[1]: *** [CMakeFiles/Makefile2:68: CMakeFiles/pc-nrfjprog-js.dir/all] Error 2
    make: *** [Makefile:84: all] Error 2
    ERR! OMG Process terminated: 2
    Unhandled rejection Error: Process terminated: 2
        at ChildProcess.<anonymous> (/home/m/git/cloud-iris/gateway/js/common/node_modules/cmake-js/lib/cMake.js:372:28)
        at emitTwo (events.js:106:13)
        at ChildProcess.emit (events.js:191:7)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:219:12)

</details>


Do I need a certain compiler version?

--- a/cmake/gcc.cmake
+++ b/cmake/gcc.cmake
@@ -12,7 +12,6 @@ add_compile_options(
 )
 
 add_compile_options(
-    -std=c++11
     -Wlogical-op
 )
 

--- a/src/nrfjprogjs.h
+++ b/src/nrfjprogjs.h
@@ -36,12 +36,18 @@
 #ifndef __NRFJPROG_H__
 #define __NRFJPROG_H__
 
+#if __cplusplus <= 199711L
+  #error This library needs at least a C++11 compliant compiler
+#endif
+
 #include <nan.h>
 #include "common.h"
 #include "dllfunc.h"
 #include "osfiles.h"
 
 #include "utility/errormessage.h"
+
+#include <functional>
 
 class Baton;
 

--- a/src/nrfjprogjs.h
+++ b/src/nrfjprogjs.h
@@ -36,10 +36,6 @@
 #ifndef __NRFJPROG_H__
 #define __NRFJPROG_H__
 
-#if __cplusplus <= 199711L
-  #error This library needs at least a C++11 compliant compiler
-#endif
-
 #include <nan.h>
 #include "common.h"
 #include "dllfunc.h"

